arcgis-ios-sdk-samples/Display information/Sketch on the map/SketchViewController.swift

```diff
@@ -16,76 +16,93 @@ import UIKit
-        
-        NotificationCenter.default.addObserver(self, selector: #selector(SketchViewController.respondToGeomChanged), name: .AGSSketchEditorGeometryDidChange, object: nil)
-        
-        // set viewpoint
-        self.mapView.setViewpoint(AGSViewpoint(targetExtent: AGSEnvelope(xMin: -10049589.670344, yMin: 3480099.843772, xMax: -10010071.251113, yMax: 3512023.489701, spatialReference: .webMercator())))
-    }
-    
-    @objc
-    func respondToGeomChanged() {
-        // Enable/disable UI elements appropriately
-        self.undoBBI.isEnabled = self.sketchEditor.undoManager.canUndo
-        self.redoBBI.isEnabled = self.sketchEditor.undoManager.canRedo
-        self.clearBBI.isEnabled = self.sketchEditor.geometry != nil && !self.sketchEditor.geometry!.isEmpty
     }
     
     // MARK: - Actions
     
-    @IBAction func geometryValueChanged(_ segmentedControl: UISegmentedControl) {
-        switch segmentedControl.selectedSegmentIndex {
-        case 0:// point
-            self.sketchEditor.start(with: nil, creationMode: .point)
-            
-        case 1:// polyline
-            self.sketchEditor.start(with: nil, creationMode: .polyline)
-            
-        case 2:// freehand polyline
-            self.sketchEditor.start(with: nil, creationMode: .freehandPolyline)
-            
-        case 3:// polygon
-            self.sketchEditor.start(with: nil, creationMode: .polygon)
-            
-        case 4:// freehand polygon
-            self.sketchEditor.start(with: nil, creationMode: .freehandPolygon)
-            
-        default:
-            break
+    @IBAction func addGeometryButtonTapped(_ sender: UIBarButtonItem) {
+        // Create an alert controller for the action sheets.
+        let alertController = UIAlertController(title: "Select a creation mode", message: nil, preferredStyle: .actionSheet)
+        // Create an action for each creation mode and add it to the alert controller.
+        creationModes.forEach { name, mode in
+            let action = UIAlertAction(title: name, style: .default) { (_) in
+                self.statusLabel.text = "\(name) selected."
+                self.sketchEditor.start(with: nil, creationMode: mode)
+            }
+            alertController.addAction(action)
         }
+        // Add "cancel" item.
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertController.addAction(cancelAction)
         
-        self.mapView.sketchEditor = self.sketchEditor
+        // Present the action sheets when the add button is tapped.
+        alertController.popoverPresentationController?.barButtonItem = addBarButtonItem
+        present(alertController, animated: true)
     }
     
     @IBAction func undo() {
-        if self.sketchEditor.undoManager.canUndo { // extra check, just to be sure
+        // Check if there are actions to undo.
+        if self.sketchEditor.undoManager.canUndo {
             self.sketchEditor.undoManager.undo()
         }
```

Comment:
Explicit `self` isn't needed. Also, this would be a good place for a `guard` statement.

```suggestion
        guard sketchEditor.undoManager.canUndo else { return }
        sketchEditor.undoManager.undo()
```